File: testdata/polyglot/swift/Vuln.swift
// Deliberately vulnerable appsec fixture. Every issue is planted. DO NOT fix.
// Never compiled; exists only to be scanned by semgrep.
//
// Swift LANDED via the curated local ruleset (argus/curated): the registry
// packs (p/swift, p/default) caught none of the plants below, so each is
// covered by a hand-written rule in internal/scanner/rules/curated.yaml,
// proven by TestProfileRecall. .swift counts as SAST-covered in skip
// accounting. Safe-code counterparts live in fp/safe.swift.

import Foundation
import CommonCrypto

func takeInput() -> String {
    return CommandLine.arguments.count > 1 ? CommandLine.arguments[1] : ""
}

// PLANT(swift-sqli, min-profile=standard, CWE-89): SQL built by string interpolation (caught by argus/curated)
func sqli(userInput: String, db: OpaquePointer?) {
    let query = "SELECT * FROM users WHERE name = '\(userInput)'"
    var stmt: OpaquePointer?
    sqlite3_prepare_v2(db, query, -1, &stmt, nil)
}

// PLANT(swift-weak-hash, min-profile=standard, CWE-328): MD5 over sensitive input (caught by argus/curated)
func weakHash(userInput: String) -> [UInt8] {
    var digest = [UInt8](repeating: 0, count: Int(CC_MD5_DIGEST_LENGTH))
    let data = Array(userInput.utf8)
    CC_MD5(data, CC_LONG(data.count), &digest)
    return digest
}

// PLANT(swift-tls-verify, min-profile=standard, CWE-295): TLS certificate validation disabled (caught by argus/curated)
final class TrustAll: NSObject, URLSessionDelegate {
    func urlSession(_ session: URLSession,
                    didReceive challenge: URLAuthenticationChallenge,
                    completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
        let cred = URLCredential(trust: challenge.protectionSpace.serverTrust!)
        completionHandler(.useCredential, cred)
    }
}

// PLANT(swift-cmdi, min-profile=standard, CWE-78): Process invoking a shell with concatenated input (caught by argus/curated)
func runShell(userInput: String) {
    let task = Process()
    task.launchPath = "/bin/sh"
    task.arguments = ["-c", "echo " + userInput]
    try? task.run()
}

// PLANT(swift-hardcoded-secret, min-profile=standard, CWE-798): hardcoded credential (caught by argus/curated)
let apiKey = "AKIAIOSFODNN7EXAMPLE"

func main() {
    let input = takeInput()
    sqli(userInput: input, db: nil)
    _ = weakHash(userInput: input)
    runShell(userInput: input)
    _ = apiKey
}

File: testdata/polyglot/python/vuln.py
# Deliberately vulnerable appsec fixture. Every issue is planted. DO NOT fix.
import hashlib
import os
import sqlite3
import subprocess


def sql_injection(username):
    conn = sqlite3.connect("app.db")
    cur = conn.cursor()
    # PLANT(py-sqli, min-profile=max, CWE-89): SQL injection via string formatting
    cur.execute("SELECT * FROM users WHERE name = '%s'" % username)
    return cur.fetchall()


def command_injection(host):
    # PLANT(py-cmdi, min-profile=standard, CWE-78): OS command injection via shell=True with concatenation
    return subprocess.check_output("ping -c 1 " + host, shell=True)


def weak_hash(password):
    # PLANT(py-weak-hash, min-profile=standard, CWE-327): weak hash (MD5) over a password (textbook CWE-328; semgrep emits CWE-327)
    return hashlib.md5(password.encode()).hexdigest()


def path_traversal(filename):
    # PLANT(py-path-traversal, min-profile=standard, CWE-22): path traversal via unsanitized user filename joined under a base dir (caught by argus/curated)
    with open(os.path.join("/var/data", filename)) as fh:
        return fh.read()
